milli/src/search/new/ranking_rule_graph/fid/mod.rs

```diff
@@ -68,24 +71,27 @@ impl RankingRuleGraphTrait for FidGraph {
             all_fields.extend(fields);
         }
 
+        let weights_map = ctx.index.fieldids_weights_map(ctx.txn)?;
+
         let mut edges = vec![];
         for fid in all_fields.iter().copied() {
+            let weight = weights_map.weight(fid).unwrap();
             edges.push((
-                fid as u32 * term.term_ids.len() as u32,
-                conditions_interner.insert(FidCondition { term: term.clone(), fid }),
+                weight as u32 * term.term_ids.len() as u32,
+                conditions_interner.insert(FidCondition { term: term.clone(), fid: Some(fid) }),
             ));
         }
 
         // always lookup the max_fid if we don't already and add an artificial condition for max scoring
-        let max_fid: Option<u16> = ctx.index.searchable_fields_ids(ctx.txn)?.into_iter().max();
+        let max_weight: Option<u16> = weights_map.max_weight();
 
-        if let Some(max_fid) = max_fid {
-            if !all_fields.contains(&max_fid) {
+        if let Some(max_weight) = max_weight {
+            if !all_fields.contains(&max_weight) {
```

Comment:
Sure, the comment above the computation of max_weight is related to why we are doing this:

```rust
 // always lookup the max_fid if we don't already and add an artificial condition for max scoring
```

The `rankingScore` of graph-based ranking rules is computed by using the max cost of all paths in the graph. For the score to be independent from the dataset, this may require adding an "artificial edge" like here with the max theoretical cost for an edge.

For the fid rule, without that path, we would depend on the dataset, in that the max cost of the graph would depend of the fields the query terms appear in. For instance, with the fields "title", "overview" and their weights 0 and 1, if I have a query with words that only appears in "title", then `all_fields` will only contain the weight 0, and the worst path will have cost 0. Now if I add a document with these words in the "overview" field, suddenly the `all_fields` contains 1, and the worst path becomes `1 * words_in_query.len()`.

The artificial edge would trigger in the first example, ensuring that the worst path is always `1 * words_in_query.len()`.